x-pack/plugins/security_solution/public/flyout/right/tabs/table_tab.tsx

```diff
@@ -7,9 +7,90 @@
 
 import type { FC } from 'react';
 import React, { memo } from 'react';
-import { TimelineTabs } from '../../../../common/types';
-import { EventFieldsBrowser } from '../../../common/components/event_details/event_fields_browser';
+import { EuiText } from '@elastic/eui';
+import { get } from 'lodash';
+import memoizeOne from 'memoize-one';
+import type { EventFieldsData } from '../../../common/components/event_details/types';
+import { FieldValueCell } from '../../../common/components/event_details/table/field_value_cell';
+import type { BrowserField, BrowserFields } from '../../../../common/search_strategy';
+import { FieldNameCell } from '../../../common/components/event_details/table/field_name_cell';
+import {
+  CellActionsMode,
+  SecurityCellActions,
+  SecurityCellActionsTrigger,
+} from '../../../common/components/cell_actions';
+import { getSourcererScopeId } from '../../../helpers';
+import * as i18n from '../../../common/components/event_details/translations';
 import { useRightPanelContext } from '../context';
+import type { ColumnsProvider } from '../../../common/components/event_details/event_fields_browser';
+import { EventFieldsBrowser } from '../../../common/components/event_details/event_fields_browser';
+import { TimelineTabs } from '../../../../common/types';
+
+export const getFieldFromBrowserField = memoizeOne(
+  (keys: string[], browserFields: BrowserFields): BrowserField | undefined =>
+    get(browserFields, keys),
+  (newArgs, lastArgs) => newArgs[0].join() === lastArgs[0].join()
+);
+
+export const getColumns: ColumnsProvider = ({
+  browserFields,
+  eventId,
+  contextId,
+  scopeId,
+  getLinkValue,
+  isDraggable,
+}) => [
+  {
+    field: 'field',
+    name: (
+      <EuiText size="xs">
+        <strong>{i18n.FIELD}</strong>
+      </EuiText>
+    ),
+    render: (field, data) => {
+      return (
+        <FieldNameCell data={data as EventFieldsData} field={field} fieldMapping={undefined} />
+      );
+    },
+  },
+  {
+    field: 'values',
+    name: (
+      <EuiText size="xs">
+        <strong>{i18n.VALUE}</strong>
+      </EuiText>
+    ),
+    render: (values, data) => {
+      const fieldFromBrowserField = getFieldFromBrowserField(
+        [data.category as string, 'fields', data.field],
+        browserFields
+      );
+      return (
+        <SecurityCellActions
+          data={{
```

Comment:
I'm not sure about the implementation of the table and how it memoizes rows/columns. Would these inlined objects result in re-renders? Or does the memoization happen in a parent component?